### PR TITLE
feat: use localized fake pattern in street

### PR DESCRIPTION
**Pull request by ST-DDT**
This PR changes the `address.street` method to use the `address.street` locale data with fake to generate street names.

For that I checked, that all street files are named correctly (street -> patterns; street _name -> actual street names).
I did not check whether the specified patterns make sense or the street names are real.

I generated the missing street files with the original pattern from the method to ensure it does not break between versions.
I did not add `'{{address.street_name}}'` to the street patterns if both the street and street_name files exist .
I did not remove it from the patterns if the street_name file is missing in that locale.

Everything in the first commit was generated by using this code:

<details>
<summary>generateLocales.ts</summary>

````ts
const processed = new Set<string>();

/**
 * Use this hook method to selectively update locale data files (not for index.ts files).
 * This method is intended to be temporarily overwritten for one-time updates.
 *
 * @param filePath The full file path to the file.
 * @param locale The locale for that file.
 * @param localePath The locale path parts (after the locale).
 */
function updateLocaleFileHook(
  filePath: string,
  locale: string,
  localePath: string[]
): void {
  if (processed.has(locale)) {
    return;
  }
  if (localePath[0] !== 'address') {
    return;
  }
  processed.add(locale);

  let streetPattern: string[];
  let streetNames: string[];

  const streetFiles = [
    'street',
    'streets',
    'street_name',
    'street_root',
    'street_title',
  ];
  for (const name of streetFiles) {
    if (existsSync(`src/locales/${locale}/address/${name}.ts`)) {
      let content: string[] =
        require(`../src/locales/${locale}/address/${name}`).default;
      unlinkSync(resolve(`src/locales/${locale}/address/${name}.ts`));
      if (content[0].includes('{{')) {
        for (const replaced of streetFiles) {
          content = content.map((e) =>
            e.replace(
              new RegExp(`{{address.${replaced}}}`, 'g'),
              '{{address.street_name}}'
            )
          );
        }
        streetPattern = content;
      } else {
        streetNames = content;
      }
    }
  }

  if (streetPattern == null) {
    if (streetNames == null) {
      streetPattern = [
        '{{name.firstName}} {{address.streetSuffix}}',
        '{{name.lastName}} {{address.streetSuffix}}',
      ];
    } else {
      streetPattern = [
        '{{name.firstName}} {{address.streetSuffix}}',
        '{{name.lastName}} {{address.streetSuffix}}',
        '{{address.street_name}}',
      ];
    }
  }

  let streetPatternContent = `export default ${JSON.stringify(
    streetPattern,
    undefined,
    '  '
  )};\n`;
  streetPatternContent = formatTypescript(streetPatternContent);
  writeFileSync(
    resolve(`src/locales/${locale}/address/street.ts`),
    streetPatternContent
  );

  if (streetNames != null) {
    let streetNamesContent = `export default ${JSON.stringify(
      streetNames,
      undefined,
      '  '
    )};\n`;
    streetNamesContent = formatTypescript(streetNamesContent);

    writeFileSync(
      resolve(`src/locales/${locale}/address/street_name.ts`),
      streetNamesContent
    );
  }

  if (existsSync(`src/locales/${locale}/address/street_address.ts`)) {
    const content: AddressDefinitions['street_address'] =
      require(`../src/locales/${locale}/address/street_address`).default;
    content.normal = content.normal
      .replace(/{{address.streetName}}/g, '{{address.street}}')
      .replace(/{{address.street_name}}/g, '{{address.street}}');
    content.full = content.full
      .replace(/{{address.streetName}}/g, '{{address.street}}')
      .replace(/{{address.street_name}}/g, '{{address.street}}');
    let streetAddressContent = `export default ${JSON.stringify(
      content,
      undefined,
      '  '
    )};\n`;
    streetAddressContent = formatTypescript(streetAddressContent);

    writeFileSync(
      resolve(`src/locales/${locale}/address/street_address.ts`),
      streetAddressContent
    );
  }
}
````

</details>

And executed it using this helper:

````sh
untilSuccess() {
        until "$@"; do :; done
}

untilSuccess pnpm run generate:locales
pnpm run generate:locales
````

### CODE REVIEW

File: src/definitions/address.ts [hunk 1]
```diff
@@ -14,7 +14,11 @@ export type AddressDefinitions = LocaleEntry<{
   postcode: string | string[];
 
   /**
-   * Names of actual cities
+   * The patterns to generate city names.
+   */
+  city: string[];
```
Comment (ST-DDT) [inline src/definitions/address.ts:19]: I forgot to add this in #948.

File: test/scripts/apidoc/examplesAndDeprecations.spec.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import { loadProject } from './utils';
 const locales: Record<string, string> = {
   GH: 'en_GH',
   US: 'en_US',
+  DE: 'de',
 };
```
Comment (Shinigami92) [inline test/scripts/apidoc/examplesAndDeprecations.spec.ts:20]: **question:** what are these and why only these 3?
Also why was `de` added here?

**Review by Shinigami92 — APPROVED** [review]
Trust-bonus-approval 😜

File: test/scripts/apidoc/examplesAndDeprecations.spec.ts (hunk 2 above)
Comment (ST-DDT) [reply test/scripts/apidoc/examplesAndDeprecations.spec.ts:20]: https://github.com/faker-js/faker/blob/09e6248a541b7fe40a41cb93b70cdd9054deafb4/src/modules/address/index.ts#L159

This is one of the few languages the example is working in.
`en` does not have `street_names` (and thus would trigger the deprecation warning and thus fail the test).

**Review by import-brain — DISMISSED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Pull request merged**